infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -128,32 +146,77 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      *
      * @param txnContext All of the above must execute in the same transaction passed in.
      */
-    public void migrateData(TxnContext txnContext) {
+    public void migrateData(CorfuStore corfuStore, TxnContext txnContext) {
         // Currently only the LogReplicationMetadataManager needs data-migration
         LogReplicationMetadataManager.migrateData(txnContext);
-        addSnapshotSyncEventOnUpgradeCompletion(txnContext);
+        addSnapshotSyncEventOnUpgradeCompletion(corfuStore, txnContext);
     }
 
     /**
-     * Add flag to event table to trigger snapshot sync.
+     * This is a helper method that is used to construct the sessions from the existing V1 metadata
+     * instead of otherwise relying on the session manager. Intended for FULL_TABLE replication.
      */
-    public void addSnapshotSyncEventOnUpgradeCompletion(TxnContext txnContext) {
-        UUID rollingUpgradeForceSyncId = UUID.randomUUID();
-
-        // Write a rolling upgrade force snapshot sync event to the logReplicationEventTable
-        LogReplicationMetadata.ReplicationEventInfoKey key = LogReplicationMetadata.ReplicationEventInfoKey.newBuilder()
-                .build();
-
-        LogReplicationMetadata.ReplicationEvent event = LogReplicationMetadata.ReplicationEvent.newBuilder()
-                .setEventId(rollingUpgradeForceSyncId.toString())
-                .setType(LogReplicationMetadata.ReplicationEvent.ReplicationEventType.UPGRADE_COMPLETION_FORCE_SNAPSHOT_SYNC)
-                .setEventTimestamp(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build())
-                .build();
+    @VisibleForTesting
+    public List<LogReplicationSession> buildSessionsFromOldMetadata(CorfuStore corfuStore, TxnContext txnContext) {
```

Comment:
It will be good to add a comment for this method explaining how it constructs the sessions from the old metadata